File: Baekjoon/String/괄호(9012).py
import sys
input = sys.stdin.readline

result = []
N = int(input())
for n in range(N):
    PS = input().strip()

    while PS.find('()') != -1:
        PS = PS.replace('()', '')
    
    result.append('NO' if len(PS) else 'YES')

print('\n'.join(result))

File: Baekjoon/Basic/괄호(9012).js
/**
 * Parenthesis String
 * 
 * VPS: (), (()), (())()
 * 괄호가 정상적으로 닫혀있으면 VPS
 * 
 */


const rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

const input = [];

rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    const N = +input[0];
    const result = [];
    for (let n=1; n<N+1; n++) {
        let PS = input[n];
        while (PS.includes('()')) {
            PS = PS.replace(/\(\)/g, '');
        }
        result.push(PS.length ? 'NO' : 'YES');
    }
    console.log(result.join('\n'));
});